data/yam/agama/auto/template.jsonnet

```diff
@@ -0,0 +1,20 @@
+local base_lib = import 'lib/base.libsonnet';
+local scripts = import 'lib/scripts.libsonnet';
+local storage = import 'lib/storage.libsonnet';
+
+function(storage_schema='', product='SLES', pre_scripts='',
+  post_scripts='', encrypted=false) {
```

Comment:
perhaps this could be `scripts_pre` so we don't need to invent names, but to follow outer to inner the structure and use that as parameter.